MapboxDirections/MBRouteStep.swift

```diff
@@ -547,7 +547,7 @@ public class RouteStep: NSObject, NSSecureCoding {
         self.coordinates = coordinates
     }
     
-    internal convenience init(json: JSONDictionary) {
+    public convenience init(json: [ String: Any ]) {
```

Comment:
Change `[String: Any]` back to `JSONDictionary`: in this library, `JSONDictionary` is a typedef for `[String: AnyObject]`. JSON objects are represented as `[String: AnyObject]` in Swift 2.3 but as `[String: Any]` in Swift 3.

This is why all the builds are failing.